src/ground_vehicle.cpp

```diff
@@ -309,7 +309,13 @@ GroundVehicleAcceleration GroundVehicle<T, Type>::GetAcceleration()
 				ShowTrainTooHeavyAdviceMessage(this);
 			}
 
-			if (Train::From(this)->UsingRealisticBraking() && _settings_game.vehicle.limit_train_acceleration) {
+			bool using_realistic_braking = Train::From(this)->UsingRealisticBraking();
+
+			if (using_realistic_braking) {
+				accel = (accel * _settings_game.vehicle.train_acc_braking_percent) / 100;
+			}
```

Comment:
In braking, the setting adjusts the maximum deceleration, but in acceleration, it adjusts the calculated value.

Should the clamp for the limit_train_acceleration setting also take into account this setting?

If accel is small and train_acc_braking_percent is < 100, the result could end up as 0 which is best avoided as it could lead to trains becoming completely stuck going uphill. It may be better to use DivTowardsPositiveInf or similar here.